jina/types/document/__init__.py

```diff
@@ -315,7 +321,7 @@ def update(self, source: 'Document',
 
         if exclude_fields is None:
             if include_fields:
-                exclude_fields = tuple(f for f in self.non_empty_fields if f not in include_fields)
+                exclude_fields = set(f for f in self.non_empty_fields if f not in include_fields)
             else:
                 exclude_fields = self.non_empty_fields
```

Comment:
Not sure, if that is a problem, but in the if, the type of `excluded_fields` is `Set[..]` and in the else it is `Tuple[..]`.

Reply:
oh yes, I forgot to remove this, I tried to change to set but some tests failed. Will revert